### PR TITLE
feat: add method to check for permission

### PR DESCRIPTION
**Pull request by arthurdenner**
## What kind of change does this PR introduce?
Feature.

## What is the current behavior?
No current behavior.

## What is the new behavior?
There is a method to check if we have permission for the camera.

*PS: Would you prefer to return an enum instead of a boolean?*

## Notes
It's my first time writing native code. Please let me know if I missed anything or if the code can be improved. Also, I don't have an iOS device and the plugin doesn't work on the simulator, so I couldn't test it properly.

### CODE REVIEW

File: ios/Classes/SwiftFlutterBarcodeScannerPlugin.swift [hunk 1]
```diff
@@ -56,76 +56,81 @@ public class SwiftFlutterBarcodeScannerPlugin: NSObject, FlutterPlugin, ScanBarc
...
-        
-        if let scanModeReceived = args["scanMode"] as? Int {
-            if scanModeReceived == ScanMode.DEFAULT.index {
-                SwiftFlutterBarcodeScannerPlugin.scanMode = ScanMode.QR.index
-            }else{
-                SwiftFlutterBarcodeScannerPlugin.scanMode = scanModeReceived
+        if(call.method == "checkCameraPermission") {
+            result(checkForCameraPermission())
+        } else {
+            let args:Dictionary<String, AnyObject> = call.arguments as! Dictionary<String, AnyObject>;
```
Comment (arthurdenner) [inline ios/Classes/SwiftFlutterBarcodeScannerPlugin.swift]: Nothing was changed here, I just moved everything to inside the `else` block.

**Comment by arthurdenner** [comment]
@AmolGangadhare, sorry for bothering you, can you take a look at this, please?

**Comment by AmolGangadhare** [comment]
@arthurdenner Thanks for the PR.

I didn't have a chance to look at this so sorry for that.

As this PR is related to enhancing runtime permissions, but we already have the runtime permissions handled in the native part that is in Android and iOS native parts. So I'm not sure that we need this now. Still, I will manage some time to check this out and let you know about this.

**Comment by arthurdenner** [comment]
@AmolGangadhare, the PR aims to expose a method to check if we have permission granted only, not to request it. Currently, if one needs to know if we have the permission, would need to use a different package.

To give you some context, in my use case, when the user clicks a button, I call `scanBarcode`, which requests permission, etc - as you mentioned. But for a better UX, I'd like to know if we have the permission granted (aka if the user has clicked that button before) so that I can display a dialog explaining the usage of the button.

Hope that clear things out. Let me know if it doesn't.

**Comment by AmolGangadhare** [comment]
@arthurdenner 

Sorry for the delayed response. Actually I think it's redundant in nature as the permissions are already handled in native part. What I understand that you need a confirmation that the user has denied the permissions while opening barcode scanning. Right?

If I'm right then we can simplify the process. Instead of returning "-1" we can return something else to decide whether the user has denied the permissions.

Please confirm the same. My opinion is creating a separate method just to check permissions is not a proper solution to this.

P.S.: Currently flutter has released v1.12.13+hotfix.5 which is a little bit buggy (opinion based on what I experienced). I'm currently working on plugin migration for the Android module to support new APIs. 
We have to wait for some time until the migration is completed.

Thanks

**Comment by arthurdenner** [comment]
Hi, thanks for checking it out.

I don't think this PR is redundant because I'm aiming for one thing: I want to check the permission only, not interact with the camera at all. If I call any current method, the camera will pop up.

Other packages that handles permission have a similar thing, e.g. [location_permissions](https://pub.dev/packages/location_permissions#checking-permission) and [location](https://pub.dev/packages/location). Currently, I'm using [permission_handler](https://pub.dev/packages/permission_handler#checking-permission) just to cover my use case, which could be replaced with the proposed implementation.

**Pull request closed without merging**